.github/workflows/build-and-push-main.yml

```diff
@@ -0,0 +1,27 @@
+name: Build and Push
+
+on:
+  push:
+    branches: 
+      - main
+  pull_request:
+    branches: 
+      - main
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Build Docker Image
+      run: |
+        docker build -t rainbowbot:build .
+    - name: Login to GitHub Container Registry
```

Comment:
Logging into the container registry should be the first step in the pipeline, uncase the login fails we dont waste any time building the image unnecessarily